gluten-core/src/main/scala/io/glutenproject/expression/ScalarSubqueryTransformer.scala

```diff
@@ -35,7 +35,8 @@ class ScalarSubqueryTransformer(plan: BaseSubqueryExec, exprId: ExprId,
     // the first column in first row from `query`.
     val rows = query.plan.executeCollect()
     if (rows.length > 1) {
-      sys.error(s"more than one row returned by a subquery used as an expression:\n${query.plan}")
+      throw new IllegalStateException(
+        s"more than one row returned by a subquery used as an expression:\n${query.plan}")
```

Comment:
Vanilla Spark throw IllegalStateException but `sys.error` throw RuntimeException

Reply:
This changes is due to this pr removes `FallbackOneRowRelation`, queries are offlaoding to Gluten in some tests.